CIScripts/Test/Tests/VRouterAgentTests.ps1

```diff
@@ -1073,31 +1063,37 @@ function Test-VRouterAgentIntegration {
             $Message = "Even diamonds require polishing."
 
             Write-Host "    Setting up a listener on container $Container2Name..."
-            $JobListener = Invoke-Command -Session $Session2 -AsJob -ScriptBlock {
+            Invoke-Command -Session $Session2 -ScriptBlock {
                 $Command = (`
                     '$IpEndPoint = New-Object System.Net.IPEndPoint([System.Net.IPAddress]::Any, {0});' +`
                     '$UdpClient = New-Object System.Net.Sockets.UdpClient {0};' +`
                     '$Task = $UdpClient.ReceiveAsync();' +`
+                    '[System.Threading.EventWaitHandle]::new($False, [System.Threading.EventResetMode]::AutoReset, \"Listening\").Set() | Out-Null;' +`
                     '$Task.Wait();' +`
                     '$ReceivedMessage = [System.Text.Encoding]::UTF8.GetString($Task.Result.Buffer);' +`
                     'return $ReceivedMessage;') -f $Using:Port
 
-                & docker exec $Using:Container2Name powershell -Command $Command
+                $JobListener = Start-Job -ScriptBlock {
+                    Param ($ContainerName, $Command)
+                    & docker exec $ContainerName powershell -Command $Command
+                } -ArgumentList $Using:Container2Name, $Command
             }
 
-            Start-Sleep -Seconds 8
+            Wait-RemoteEvent -Session $Session2 -ContainerName $Container2Name -EventName "Listening"
```

Comment:
Does UdpClient actually listen when ReceiveAsync() returns?

Seemingly the only thing we should rely on is that returned Task object is valid.
I suspect that we actually may still rely on delay between calling ReceiveAsync() and sending UDP packet. It's now implicit though. However, additional event is for sure a good step.

I suggest adding a comment about this uncertainty - if it gets flaky, we might add an explicit delay anyway.